pylint/checkers/utils.py

```diff
@@ -416,10 +420,12 @@ def next_char(i):
                 raise UnsupportedFormatCharacter(i)
             if key:
                 keys.add(key)
+                key_types[key] = char
             elif char != '%':
                 num_args += 1
+                pos_types.append(char)
         i += 1
-    return keys, num_args
+    return keys, num_args, key_types if keys else pos_types
```

Comment:
It seems to me that we always assume the third value from this function is a dictionary (`key_types`). Why do we also need to return `pos_types` here? It's also not intuitive on why the third value can have different types, so if we actually need `pos_types`, can't we return it as a fourth value?

Reply:
Yes, `pos_types` is needed for position arguments. I've split operand as you suggested.